src/adapter/src/catalog/transact.rs

```diff
@@ -2117,250 +1850,94 @@ impl Catalog {
-                tx.update_cluster(id, cluster.clone().into())?;
-                builtin_table_updates.extend(
-                    state.resolve_builtin_table_updates(state.pack_cluster_update(&name, 1)),
-                );
+                tx.update_cluster(id, cluster.into())?;
                 info!("update cluster {}", name);
             }
             Op::UpdateItem { id, name, to_item } => {
-                builtin_table_updates
-                    .extend(state.resolve_builtin_table_updates(state.pack_item_update(id, -1)));
-                Self::update_item(
-                    state,
-                    builtin_table_updates,
-                    id,
-                    name.clone(),
-                    to_item.clone(),
-                );
-                let entry = state.get_entry(&id);
-                tx.update_item(id, entry.clone().into())?;
+                let mut entry = state.get_entry(&id).clone();
+                entry.name = name.clone();
+                entry.item = to_item.clone();
+                tx.update_item(id, entry.into())?;
 
                 if Self::should_audit_log_item(&to_item) {
-                    let name = Self::full_name_detail(
+                    let mut full_name = Self::full_name_detail(
                         &state.resolve_full_name(&name, session.map(|session| session.conn_id())),
                     );
+                    full_name.item = name.item;
 
-                    state.add_to_audit_log(
+                    CatalogState::add_to_audit_log(
+                        &state.system_configuration,
                         oracle_write_ts,
                         session,
                         tx,
-                        builtin_table_updates,
                         audit_events,
                         EventType::Alter,
                         catalog_type_to_audit_object_type(to_item.typ()),
                         EventDetails::UpdateItemV1(mz_audit_log::UpdateItemV1 {
                             id: id.to_string(),
-                            name,
+                            name: full_name,
                         }),
                     )?;
                 }
+
+                Self::log_update(state, &id);
             }
             Op::UpdateStorageUsage {
                 shard_id,
                 size_bytes,
                 collection_timestamp,
             } => {
-                state.add_to_storage_usage(
-                    tx,
-                    builtin_table_updates,
-                    shard_id,
-                    size_bytes,
-                    collection_timestamp,
+                let id = tx.get_and_increment_id(
+                    mz_catalog::durable::STORAGE_USAGE_ID_ALLOC_KEY.to_string(),
                 )?;
+                let details =
+                    VersionedStorageUsage::new(id, shard_id, size_bytes, collection_timestamp);
+                tx.insert_storage_usage_event(details);
```

Comment:
Definitely not blocking, but what do you think about moving this ID allocation into `Transaction::insert_storage_usage_event`? All of the other ID allocations seem to happen in their relevant `tx.insert_*` function, and as-is `insert_storage_usage_event` is quite shallow, so it seems like it might be a nice place to stick this extra logic.

Reply:
Good call, I've update with this change.